tests/drivers/tee/tee/src/main.c

```
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2023 Epam Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @addtogroup t_tee_driver
+ * @{
+ * @defgroup t_tee_driver test_tee_driver
+ * @}
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+#include <zephyr/device.h>
+
+#include <zephyr/drivers/tee.h>
+
+ZTEST(tee_test_suite, test_basic_calls)
+{
+	int ret;
+	uint32_t session_id;
+	int addr;
+	struct tee_shm *shm = NULL;
+	const struct device *const dev = DEVICE_DT_GET_ONE(test_tee);
+
+	zassert_not_null(dev, "Unable to get dev");
+
+	ret = tee_get_version(dev, NULL);
```

Is this correct? IMO it should fail of pointer to output structure is not present.

---

I've made 2 set of the tests. The first one is related to generic part. So I add dummy driver with compatible = test,test. It does nothing and ignores output structure. The intention was to test generic interface.

---

So what exactly is tested? That you can call a function?

---

Yes. That it generates syscall correctly.

---

Lots of zephyr unitests check that syscalls are working.

I don't like this tests because they contradict GP TEE API specifications. Basically, for anyone reading this, it looks like you are testing to ensure that your TEE API is not compatible with GP spec. 

Remember that unit tests often are used as code examples. So code under tests should always behave to spec.

---

It may sound funny, but those tests were used as code example. It shows how dummy optee driver can be implemented. This helped a lot for the team while developing optee_client. I can drop those tests is you think they are useless.

---

I will drop commit with this tests.